text/0000-build-std.md

```diff
@@ -0,0 +1,4582 @@
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why allow building from the sysroot with implied `RUSTC_BOOTSTRAP`?*][rationale-implied-bootstrap]
+
+### Panic strategies
+[panic-strategies]: #panic-strategies-1
+
+Panic strategies are unlike other profile settings insofar as they influence
+which crates and flags are passed to the standard library. For example, if
+`panic = "unwind"` were set in the Cargo profile then the `panic_unwind` feature
+would need to be provided to `std` and `-Cpanic=unwind` passed to suggest that
+the compiler use that panic runtime.
+
+If the current crate has no dependency on `std` (i.e. have added an `alloc` or
+`core` dependency explicitly to opt-out of the implicit `std` dependency), then
+Cargo will not build either of the `panic_unwind` or `panic_abort` crates or
+pass `-Cpanic` to rustc. In this circumstance, if `panic` is set in the Cargo
+profile, then this value will be ignored and Cargo will emit a warning informing
+the user of this.
+
+If the crate does depend on `std`, then Cargo's behaviour depends on whether or
+not `panic` is set in the profile:
+
+- If `panic` is not set in the profile then unwinding may still be the default
+  for the target and Cargo will need to enable the `panic_unwind` feature to the
+  standard library just in case it is used.
+- If `panic` is set to "unwind" then the `panic_unwind` feature will be enabled
+  and `-Cpanic=unwind` will be passed.
+- If `panic` is set to "abort" then `-Cpanic=abort` will be passed.
+  - `panic_abort` is a non-optional dependency of `std` so it will always be
+    built.
+
+Tests, benchmarks, build scripts and proc macros continue to ignore the "panic"
+setting and `panic = "unwind"` is always used - which means the standard library
+needs to be recompiled again if the user is using "abort". Once
+`panic-abort-tests` is stabilised, the standard library can be built with the
+profile's panic strategy even for tests and benchmarks.
+
+Cargo will not inspect the `RUSTFLAGS` environment variable for compilation
+flags that would require additional crates to be built for compilation to
+succeed.
+
+*See the following sections for future possibilities:*
+
+- [*Avoid building `panic_unwind` unnecessarily*][future-panic_unwind]
+
+### Special object files
+[special-object-files]: #special-object-files
+
+A handful of targets require linking against special object files, such as
+`windows-gnu`, `linux-musl` and `wasi` targets. For example, `linux-musl`
+targets require `crt1.o`, `crti.o`, `crtn.o`, etc.
+
+Since [rust#76185]/[compiler-team#343], the compiler has a stable
+`-Clink-self-contained` flag which will look for special object files in
+expected locations, typically populated by the `rust-std` components. Its
+behaviour can be forced by `-Clink-self-contained=true`, but is force-enabled
+for some targets and inferred for others.
+
+Rust can start to ship `rust-self-contained-$target` components for any targets
```

Comment:
Hmmm. Why does the target name need to be contained in the component name at all? Each target can just ship its own `rust-self-contained` component.